tools/firtool/firtool.cpp

```diff
@@ -299,6 +299,10 @@ static cl::opt<std::string>
     omirOutFile("output-omir", cl::desc("file name for the output omir"),
                 cl::init(""), cl::cat(mainCategory));
 
+static cl::opt<std::string> mlirOutFile(
+    "output-mlir", cl::desc("Optional file name to output the final MLIR"),
```

Comment:
Maybe the description should specifically mention that this is in addition to the primary output file format? Otherwise, someone who is just reading the output of `firtool --help` might not understand why this exists when we already have the `--ir-*` options.

The other minor thing I am wondering is if this flag name should explicitly be `--output-final-mlir`. `--output-mlir` is probably fine, and it would make sense that it would be the final form of the MLIR representation, but I was wondering if someone might misinterpret it to be the MLIR format from some intermediate stage (e.g. after parsing FIRRTL). This I don't have strong opinions about either way, just wanted to bring that up as a possibility, since I like `--output-mlir` just fine as well.

Reply:
I actually started with output-final-mlir, but renamed it to match the OMIR naming of output-omir. I think you're right on both fronts though: I'll update the naming and help output.